File: Sources/Model/ClaudeJumpPreferenceStore.swift
import Foundation

/// How "Open thread" treats a Claude *Desktop* session: land in the Desktop
/// app (default), or resume the exact conversation in a terminal via
/// `claude --resume`. Desktop can't jump to a specific chat yet — the
/// `claude://code/<bridge-id>` route exists in its code but sits behind
/// Anthropic's server-side flag — so people who care more about "exact
/// conversation" than "my usual window" can pick the CLI here.
@MainActor
final class ClaudeJumpPreferenceStore: ObservableObject {
    static let shared = ClaudeJumpPreferenceStore()

    private static let key = "AgentIsland.claudeJumpPrefersCLI"

    @Published var prefersCLI: Bool {
        didSet { UserDefaults.standard.set(prefersCLI, forKey: Self.key) }
    }

    private init() {
        // The picker is gone from Settings (2026-07-14) — desktop app is
        // the one behavior. The stored value is ignored, not erased, so
        // the picker can return without losing anyone's old choice.
        prefersCLI = false
    }
}

File: windows/src/AgentIsland/Model/ClaudeJumpPreferenceStore.cs
using System.ComponentModel;
using AgentIsland.Core;

namespace AgentIsland.Model;

/// How "Open thread" treats a Claude *Desktop* session: land in the Desktop
/// app (default), or resume the exact conversation in a terminal via
/// `claude --resume`. Desktop can't jump to a specific chat yet — its
/// `claude://code/<bridge-id>` route sits behind Anthropic's server-side
/// flag — so people who value "exact conversation" over "my usual window"
/// can pick the CLI here. (Parity with the macOS store.)
public sealed class ClaudeJumpPreferenceStore : INotifyPropertyChanged
{
    public static ClaudeJumpPreferenceStore Shared { get; } = new();

    private const string Key = "AgentIsland.claudeJumpPrefersCli";

    private bool _prefersCli;

    public event PropertyChangedEventHandler? PropertyChanged;

    private ClaudeJumpPreferenceStore()
    {
        _prefersCli = Preferences.Get<bool?>(Key) ?? false;
    }

    public bool PrefersCli
    {
        get => _prefersCli;
        set
        {
            if (_prefersCli == value) return;
            _prefersCli = value;
            Preferences.Set(Key, value);
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(PrefersCli)));
        }
    }
}
